### DISCUSSION TITLE
Node js SQS working locally but not on AWS

### DISCUSSION THREAD

**Discussion by ihsanlatheef** [serverless/serverless#10821] [Q&A]
Hi all,

Could someone explain as to why my serverless node js example works locally on my machine with the serverless-offline package but fails to work once deployed to AWS. I do not get any errors on cloudwatch. All this app does is read a CSV file and parse it and push the contents to an SQS queue on AWS. It has two lambda functions a producer and a worker. The contents of my yaml file are as follows:

_org: orgname
app: test-queue
service: test-queue

frameworkVersion: '3'

provider:
  name: aws
  runtime: nodejs14.x

constructs:
  jobs:
    type: queue
    worker:
      handler: handler.consumer

functions:
  producer:
    handler: handler.producer
    events:
      - httpApi:
          method: post
          path: /produce
    environment:
      QUEUE_URL: ${construct:jobs.queueUrl}

plugins:
  - serverless-lift
  - serverless-offline_

**Comment by mnapoli** [comment]
Hi, I think this is related to https://github.com/getlift/lift/issues/93 which is caused by https://github.com/dherault/serverless-offline/issues/1149 Serverless offline does not support resolving CloudFormation references, i.e. it cannot resolve the `${construct:jobs.queueUrl}` variable. Which means local development cannot work here unfortunately.

**Reply by ihsanlatheef** [reply]
Does that mean, if I remove the library it would work? Thank you very much for the prompt reply by the way. 🙂

**Reply by ihsanlatheef** [reply]
I removed the last line from the yaml file (i.e.  - serverless-offline) and retried. Sadly, no change.
